### PR TITLE
Add concurrency to GitHub Actions

### PR DESCRIPTION
**Pull request by jdahm**
## Description

This should cancel expensive continuous integration when new commits are added to PRs. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

### CODE REVIEW

**Comment by jdahm** [comment]
Not quite there. We still want the build tests to run on the later commit. Hmm...

**Comment by jdahm** [comment]
bors try

File: .github/workflows/gt4py-gitlab-ci.yml [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
   push:
     branches:
     - master
     - trying
     - staging
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (havogt) [inline .github/workflows/gt4py-gitlab-ci.yml:13]: Let's remove it from here. This workflow runs only for a few seconds anyway.

Comment (havogt) [inline .github/workflows/gt4py-gitlab-ci.yml:13]:
```suggestion
```

**Review by havogt — APPROVED** [review]
Let's remove it from the gitlab job.

**Pull request merged**